### PR TITLE
CLI-13157 Dev Console broken when opened too early

### PR DESCRIPTION
**Pull request by garnold24**
einsteinK seemed to have fixed this with his PR, so i included that in this one, but also fixed an issue which arose that involved the console duplicating itself.

### CODE REVIEW

File: CoreScriptsRoot/Modules/DeveloperConsoleModule.lua [hunk 1]
```diff
@@ -2719,12 +2719,16 @@ end
 -- Permissions --
 -----------------
 do
-	local permissions;
+	local permissionsLoading, permissions = false;
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: if I'm not mistaken, I don't think permissionsLoading is initialized here, although it starts as nil, which is very similar to false.

File: CoreScriptsRoot/Modules/DeveloperConsoleModule.lua [hunk 2]
```diff
@@ -2719,12 +2719,16 @@ end
...
 	function DeveloperConsole.GetPermissions()
+		while permissionsLoading do wait() end
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: seems the only reason this is necessary is due to HttpRbxApiService. Would it be possible just to spawn a thread for this call so we don't need to debounce?

File: CoreScriptsRoot/Modules/DeveloperConsoleModule.lua [hunk 3]
```diff
@@ -2990,38 +2995,37 @@ local function onDevConsoleVisibilityChanged(isVisible)
 	end
 end
 
-local getDeveloperConsoleIsCreating = false
+local devConsoleCreating = false
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: I believe this also stems from the http call, if we can isolate that I think we can make this change a lot smaller

File: CoreScriptsRoot/Modules/DeveloperConsoleModule.lua (hunk 1 above)
Comment (garnold24) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: i think permissionsLoading gets initialized to false, while permissions is nil.

File: CoreScriptsRoot/Modules/DeveloperConsoleModule.lua (hunk 2 above)
Comment (garnold24) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: Even if we spawn a new thread, I'm pretty sure we'd need to still keep the debounce, as everytime you attempt to open the console, it's going to check permissions. If we remove the debounce it's going to keep calling all of that api stuff everytime it wants to open until it goes through and creates permissions, which i don't think is what we want to have happen.

Comment (hockeychaos) [inline CoreScriptsRoot/Modules/DeveloperConsoleModule.lua]: hmm alright, I guess permissions could change in the middle of a session, but seems unnecessary to do this more than once, as it seems to complicate the code quite a bit. I'm ok with this change for now, but we should consider moving the permission check to script start up

**Pull request merged**